File: frist/StackApp.cpp
#include <iostream>
#include <cstring>
using namespace std;

class StackApp
{
private:
    /* data */
    
    char data[10] = {-2,-2,-2,-2,-2,-2,-2,-2,-2,-2};
    char *top = &data[0];

public:
    bool isEmpty()
    {
        if ((int) *top != -1)
        {
            return false;
        }
        return true;
    }

    bool isFull()
    {
        if (*(++top) != 0)
        {
            return false;
        }
        return true;
    }

    char push(char item)
    {
        if (isFull())
        {
            cout << "Stack Overflowed" << endl;
            return 0;
        }
        ++top;
        *top = item;
        return 0;
    }

    char pop()
    {
        if (isEmpty())
        {
            return 0;
        }
        char temp = *top;
        --top;
        return temp;
    }

    char peek()
    {
        if (isEmpty())
        {
            return -1;
        }
        return  *top;
    }
};

int precedence(char ope)
{
    if (ope == '(' || ope == ')')
    {
        return 3;
    }
    if (ope == '/' || ope == '*')
    {
        return 2;
    }
    return 1;
}

bool isOperator(char ope)
{
    if (ope == '+' || ope == '-' || ope == '/' || ope == '*' || ope == '(' || ope == ')')
    {
        return true;
    }
    return false;
}

void infixTopostfix(char array[], int n)
{
    StackApp *stack = new StackApp();
   
    for (int i = 0; i < n; i++)
    {
        if (!isOperator(array[i]))
        {
            cout << (char) array[i];
        }
        else if (stack->isEmpty())
        {   
            stack->push(array[i]);
        }
        else
        {
            
            
            if (precedence((char) stack->peek()) < (char) precedence(array[i]))
            {
                stack->push(array[i]);
            }
            else
            {
                while (!stack->isEmpty())
                {
                    cout <<  (char) stack->pop();
                }
                stack->push(array[i]);
            }
        }
    }
    while (!stack->isEmpty())
    {
        cout << (char) stack->pop();
    }
    cout << endl;
}

int main(int argc, char const *argv[])
{
    string s;
    cin >> s;
    int n = s.length();
    char char_array[n];
    strcpy(char_array, s.c_str());
    infixTopostfix(char_array, n);

}

File: frist/StackApp.java
package frist;

/**
 * StackApp
 */

class Stack {
    private  int top = -1;
    private  int[] data = new int[6];

    boolean isEmpty() {
        if(top != -1) {
            return false;
        }
        return true;
    }

    boolean isFull() {
        if(top != 5) {
            return false;
        }
        return true;
    }

    int push(int item) {
        if(isFull()) {
            System.out.println("Stack Overflowed");
            return 0;
        }
        ++top;
        data[top] = item;
        return 0;
    }

    int pop() {
        if(isEmpty()) {
            System.out.println("Stack Underflowed");
            return 0;
        }
        --top;
        return 0;
    }
    
    int peek() { 
        if(isEmpty()) {
            System.out.println("Stack is Empty");
            return -1;
        }
        return data[top];
    }

    int precedence(char operator) {
        if(operator == '/' || operator == '*') {
            return 2;
        }
        return 1; 
    }
}

public class StackApp {
    public static void main(String[] args) {
        Stack app = new Stack();
        app.push(1);
        app.push(2);
        app.push(3);
        app.push(4);
        app.push(5);
        app.push(6);
        System.out.println(app.peek());
        app.push(7);
        app.pop();
        app.pop();
        app.pop();
        app.pop();
        app.pop();
        System.out.println(app.peek());
        app.pop();
        app.pop();
        app.peek();
    }
}

